includes/class-wc-comments.php

```diff
@@ -405,6 +408,21 @@ public static function get_rating_counts_for_product( &$product ) {
 
 		return $counts;
 	}
+
+	/**
+	 * Update comment type of product reviews.
+	 *
+	 * @since 3.5.0
+	 * @param array $comment_data Comment data.
+	 * @return array
+	 */
+	public static function update_comment_type( $comment_data ) {
+		if ( ! is_admin() && isset( $_POST['comment_post_ID'] ) && 'product' === get_post_type( absint( $_POST['comment_post_ID'] ) ) ) { // WPCS: input var ok, CSRF ok.
```

Comment:
This should probably check to make sure `'' === comment_data['comment_type']` so we don't overwrite some other type of product comments.